### DISCUSSION TITLE
Show Hidden Files and Folders

### DISCUSSION THREAD

**Discussion by opossalite** [cmus/cmus#1058] [General]
Hello, I'm wondering how to show hidden files and folders in the file browser. I have much of my music organized by using periods and commas to shift the order that folders are displayed in, but inside cmus all folders with a period are hidden. How can I display them? Thank you in advance.

**Comment by pgaskin** [comment]
With the default keybindings, you can press `i` to show hidden files. Alternatively, you can set the `show_hidden` option.

**Comment by williamhaley** [comment]
I have an album, [...And Out Come the Wolves](https://en.wikipedia.org/wiki/...And_Out_Come_the_Wolves), that doesn't appear in `cmus`. I'm assuming the leading periods/dots/ellipses in the name are why I'm not seeing it.

I found this discussion. I saw in the [man docs for cmus that there's a `show_hidden` option](https://github.com/cmus/cmus/blob/89f1c935cbc0a345b796507bdc52c76086ecc26b/Doc/cmus.txt#L1194) (as referenced in this thread). I must be missing something, because I can't run `show_hidden` in the cmus REPL/command palette (`Error: unknown command`).

 I also can't seem to configure that option in the `cmus/rc` file.

```
$ cat ~/.config/cmus/rc
# Bundled with cmus in /usr/share/cmus/
# set colorscheme zenburn

show_hidden true
colorscheme cyan

$ cmus
cmus: Error: unknown command

cmus: Press <enter> to continue.
```

I was using the `2.10.0` package from the [official Arch repo](https://archlinux.org/packages/extra/x86_64/cmus/). I tried building the latest `master` too, but I see the same behavior. `i` doesn't seem to change anything either.

I think I'm probably missing something obvious. For now, I renamed my album's folder/directory to `And Out Come the Wolves`, did a `clear` and `add` and I see it showing up.

**Reply by gavtroy** [reply]
That's the option, so the full command would be `toggle show_hidden` or `set show_hidden=true`. Linux convention is to treat files/directories as hidden if the first character is `.` so you might prefer to rename it anyway.

**Reply by gavtroy** [reply]
And yes, it's bound to `i` by default, but it only changes whether the files are shown in browser view (5). To add it to your library you have to give the path to the hidden directory explicitly. It won't be added in a recursive add operation regardless of the `show_hidden` option.

**Reply by williamhaley** [reply]
Gotcha. That makes sense. Thanks!
